compiler-plugin/src/main/java/io/ballerina/stdlib/http/compiler/HttpCompilerPluginUtil.java

```diff
@@ -192,22 +192,58 @@ private static void validateArrayElementTypeInReturnType(SyntaxNodeAnalysisConte
         }
         if (kind == TypeDescKind.INTERSECTION) {
             TypeSymbol typeSymbol = ((IntersectionTypeSymbol) memberTypeDescriptor).effectiveTypeDescriptor();
-            validateArrayElementTypeInReturnType(ctx, node, typeStringValue, typeSymbol, diagnosticCode);
+            validateArrayElementTypeInReturnType(ctx, node, typeStringValue, typeSymbol,
+                    diagnosticCode);
         } else if (kind == TypeDescKind.TYPE_REFERENCE) {
+            if (isStatusCodeResponseType((TypeReferenceTypeSymbol) memberTypeDescriptor)) {
+                reportInvalidReturnType(ctx, node, typeStringValue, diagnosticCode);
+                return;
+            }
             TypeSymbol typeDescriptor = ((TypeReferenceTypeSymbol) memberTypeDescriptor).typeDescriptor();
             TypeDescKind typeDescKind = retrieveEffectiveTypeDesc(typeDescriptor);
-            if (typeDescKind != TypeDescKind.RECORD && typeDescKind != TypeDescKind.MAP &&
-                    typeDescKind != TypeDescKind.TABLE) {
+            if (typeDescKind == TypeDescKind.OBJECT || typeDescKind == TypeDescKind.ERROR) {
                 reportInvalidReturnType(ctx, node, typeStringValue, diagnosticCode);
+            } else if (typeDescKind == TypeDescKind.UNION) {
+                List<TypeSymbol> typeSymbols = ((UnionTypeSymbol) typeDescriptor).userSpecifiedMemberTypes();
+                for (TypeSymbol typeSymbol : typeSymbols) {
+                    TypeDescKind effectiveTypeDesc = retrieveEffectiveTypeDesc(typeSymbol);
+                    if (effectiveTypeDesc == TypeDescKind.OBJECT || effectiveTypeDesc == TypeDescKind.ERROR) {
+                        reportInvalidReturnType(ctx, node, typeStringValue, diagnosticCode);
+                        return;
+                    } else if (effectiveTypeDesc == TypeDescKind.TYPE_REFERENCE) {
+                        if (isStatusCodeResponseType((TypeReferenceTypeSymbol) typeSymbol)) {
+                            reportInvalidReturnType(ctx, node, typeStringValue, diagnosticCode);
+                            return;
+                        }
+                        Optional<String> definitionName = ((TypeReferenceTypeSymbol) typeSymbol).definition().getName();
+                        if (definitionName.isPresent() && definitionName.get().equals("StatusCodeResponse")) {
```

Comment:
What is the purpose of this check? Does this checks the `StatusCodeResponse` included records? i.e.
```bal
type AlbumConflict record {|
    *http:Conflict;
    record {
        string message;
    } body;
|};
```

Reply:
It only checks for `StatusCodeResponse` type